addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/profiles/EnOceanProfileFactory.java

```diff
@@ -0,0 +1,99 @@
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+import org.eclipse.smarthome.core.thing.profiles.ProfileFactory;
+import org.eclipse.smarthome.core.thing.profiles.ProfileType;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeProvider;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link EnOceanProfileFactory} is responsible for creating profiles.
+ *
+ * @author Jan Kemmler - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ProfileFactory.class)
+public class EnOceanProfileFactory implements ProfileFactory, ProfileAdvisor, ProfileTypeProvider {
+
+    @SuppressWarnings("null")
+    private static final Set<ProfileTypeUID> SUPPORTED_PROFILE_TYPE_UIDS = Stream
+            .of(EnOceanProfiles.ROCKER_TO_ON_OFF, EnOceanProfiles.ROCKER_TO_DIMMER).collect(Collectors.toSet());
+
+    @SuppressWarnings("null")
+    private static final Set<ProfileType> SUPPORTED_PROFILE_TYPES = Stream
+            .of(EnOceanProfiles.ROCKER_TO_ON_OFF_TYPE, EnOceanProfiles.ROCKER_TO_DIMMER_TYPE)
+            .collect(Collectors.toSet());
+
+    @Override
+    @Nullable
+    public Profile createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback, ProfileContext context) {
+        if (EnOceanProfiles.ROCKER_TO_ON_OFF.equals(profileTypeUID)) {
+            return new RockerChannelToOnOffProfile(callback);
+        }
+        if (EnOceanProfiles.ROCKER_TO_DIMMER.equals(profileTypeUID)) {
+            return new RockerChannelToDimmerProfile(callback, context);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Return the identifiers of all supported profile types
+     *
+     * @return a collection of all profile type identifier which this class is capable of creating
+     */
+    @Override
+    public Collection<ProfileTypeUID> getSupportedProfileTypeUIDs() {
+        return SUPPORTED_PROFILE_TYPE_UIDS;
+    }
+
+    @Override
+    public Collection<ProfileType> getProfileTypes(@Nullable Locale locale) {
+        return SUPPORTED_PROFILE_TYPES;
+    }
+
+    @Override
+    public @Nullable ProfileTypeUID getSuggestedProfileTypeUID(@NonNull Channel channel, @Nullable String itemType) {
+        switch (channel.getKind()) {
+            case STATE:
+                return null;
+            case TRIGGER:
+                if (new ChannelTypeUID("enocean", "rocker_channel").equals(channel.getChannelTypeUID())) {
+                    if ("Switch".equalsIgnoreCase(itemType)) {
+                        return EnOceanProfiles.ROCKER_TO_ON_OFF;
+                    } else if ("Dimmer".equalsIgnoreCase(itemType)) {
+                        return EnOceanProfiles.ROCKER_TO_DIMMER;
+                    }
+                }
+                break;
+            default:
+                throw new NotImplementedException();
```

Comment:
I think its better to return `null` here instead of throwing an exception that is never caught.